### PR TITLE
Collateral Escrow

### PR DESCRIPTION
**Pull request by sypeer**
TLRV2-423

### CODE REVIEW

File: packages/contracts/contracts/escrow/CollateralEscrowFactory.sol [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     */
+    function deployCollateralEscrow(uint256 _bidId)
+        external
+        returns(address)
+    {
+        BeaconProxy proxy_ = new BeaconProxy(
+            _collateralEscrowBeacon,
+            abi.encodeWithSelector(CollateralEscrowV1.initialize.selector, _bidId)
+        );
+        _escrows[_bidId] = address(proxy_);
```
Comment (ethereumdegen) [inline packages/contracts/contracts/escrow/CollateralEscrowFactory.sol:41]: doesnt this mean that anyone can overwrite anyone elses ?

File: packages/contracts/contracts/escrow/CollateralEscrowV1.sol [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+            IERC721Upgradeable(_collateralAddress)
+                .safeTransferFrom(
+                    msg.sender,
+                    address(this),
+                    _tokenId
+                );
+            return;
+        }
+        // Deposit ERC1155
+        if (_collateralType == CollateralType.ERC721) {
```
Comment (ethereumdegen) [inline packages/contracts/contracts/escrow/CollateralEscrowV1.sol:132]: should say ERC1155

File: packages/contracts/contracts/escrow/CollateralEscrowV1.sol [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+            SafeERC20Upgradeable.safeTransferFrom(
+                IERC20Upgradeable(_collateralAddress),
+                msg.sender,
+                address(this),
+                _amount
+            );
+            return;
+        }
+        // Deposit ERC721
+        if (_collateralType == CollateralType.ERC721) {
```
Comment (ethereumdegen) [inline packages/contracts/contracts/escrow/CollateralEscrowV1.sol:122]: should enforce that amount is 1

File: packages/contracts/contracts/escrow/CollateralEscrowV1.sol [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+     * @param _amount The amount to withdraw.
+     * @param _recipient The address to send the assets to.
+     */
+    function withdraw(address _collateralAddress, uint256 _amount, address _recipient)
+        external
+        onlyOwner
+    {
+        require(_amount > 0, "Withdraw amount cannot be zero");
+        Collateral memory collateral = _balances[_collateralAddress];
+        _withdrawCollateral(collateral, _collateralAddress, _amount, _recipient);
```
Comment (ethereumdegen) [inline packages/contracts/contracts/escrow/CollateralEscrowV1.sol:100]: should modify the balances mapping !

File: packages/contracts/contracts/escrow/CollateralEscrowFactory.sol (hunk 1 above)
Comment (sypeer) [reply packages/contracts/contracts/escrow/CollateralEscrowFactory.sol:41]: removed the escrow factory and replaced with a library in https://github.com/teller-protocol/teller-protocol-v2/pull/6

**Pull request closed without merging**